### PR TITLE
Button: make sure buttont color takes precedence over text color.

### PR DESCRIPTION
**Pull request by alansouzati**
#### What does this PR do?
Changes button to fix an issue where the theme color was being overwritten by the text color
#### Where should the reviewer start?
StyledButton
#### What testing has been done on this PR?
manual
#### How should this be manually tested?
open Themed Color storybook example in
#### What are the relevant issues?
Fixes https://github.com/grommet/grommet/issues/2396
#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
yes
#### Is this change backwards compatible or is it a breaking change?
backwards

### CODE REVIEW

File: src/js/components/Button/StyledButton.js [hunk 1]
```diff
@@ -99,8 +99,8 @@ export const StyledButton = styled.button`
 
   ${genericStyles}
   ${props => props.plain && plainStyle}
-  ${props => !props.plain && basicStyle(props)}
   ${props => props.primary && primaryStyle(props)}
+  ${props => !props.plain && basicStyle(props)}
```
Comment (ericsoderberghp) [inline src/js/components/Button/StyledButton.js]: This doesn't look right to me. We should be applying the basicStyle first and then adjusting it based on finer grained properties. What about having primaryStyle second was causing an issue? I would think that's where we should make a change.

Comment (alansouzati) [reply src/js/components/Button/StyledButton.js]: `primaryStyle` sets background and color again (through `backgroundStyle` function). `color` is set by `basicStyle` and then primaryStyle overrides it even thought `button.color` is set.

File: src/js/components/Button/button.stories.js [hunk 2]
```diff
@@ -149,6 +150,32 @@ const ColoredButton = props => (
   </Grommet>
 );
 
+const customButtonColor = deepMerge(grommet, {
```
Comment (ericsoderberghp) [inline src/js/components/Button/button.stories.js]: I think `Grommet` is already doing a `deepMerge`?

File: src/js/utils/background.js [hunk 3]
```diff
@@ -43,19 +43,20 @@ export const backgroundIsDark = (backgroundArg, theme) => {
   return result;
 };
 
-export const backgroundStyle = (backgroundArg, theme) => {
+export const backgroundStyle = (backgroundArg, theme, componentTextColor) => {
```
Comment (ericsoderberghp) [inline src/js/utils/background.js]: I'd probably just call this `textColor`. I don't think the `component` prefix is needed, just like it isn't `componentBackgroundArg`.

File: src/js/components/Button/button.stories.js (hunk 2 above)
Comment (alansouzati) [reply src/js/components/Button/button.stories.js]: Grommet does deepMerge of the base theme. I want the font to be the grommet one. and enhance the rest. makes sense ?

File: src/js/utils/background.js (hunk 3 above)
Comment (alansouzati) [reply src/js/utils/background.js]: maybe textArg ? to align better. textColor is already being used.

Comment (ericsoderberghp) [reply src/js/utils/background.js]: I'd choose the argument names first and then decide on internal variable names.

File: src/js/components/Button/button.stories.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/Button/button.stories.js]: Ah. Thanks.

**Comment by djalmaaraujo** [comment]
🎉

**Pull request merged**